foxtrot-core/src/main/java/com/flipkart/foxtrot/core/alerts/EmailBuilder.java

```diff
@@ -0,0 +1,89 @@
+package com.flipkart.foxtrot.core.alerts;
+
+import com.flipkart.foxtrot.core.email.Email;
+import com.flipkart.foxtrot.core.email.RichEmailBuilder;
+import com.flipkart.foxtrot.core.exception.FoxtrotException;
+import com.flipkart.foxtrot.core.internalevents.InternalSystemEvent;
+import com.flipkart.foxtrot.core.internalevents.InternalSystemEventVisitor;
+import com.flipkart.foxtrot.core.internalevents.events.QueryProcessed;
+import com.flipkart.foxtrot.core.internalevents.events.QueryProcessingError;
+import com.google.common.base.CaseFormat;
+
+import java.util.Collections;
+
+/**
+ *
+ */
+class EmailBuilder implements InternalSystemEventVisitor<Email> {
+
+    private final RichEmailBuilder richEmailBuilder;
+
+    EmailBuilder(RichEmailBuilder richEmailBuilder) {
+        this.richEmailBuilder = richEmailBuilder;
+    }
+
+
+    @Override
+    public Email visit(QueryProcessed queryProcessed) {
+        return null;
+    }
+
+    @Override
+    public Email visit(QueryProcessingError queryProcessingError) {
+        final FoxtrotException exception = queryProcessingError.getException();
+        switch (exception.getCode()) {
+            case CARDINALITY_OVERFLOW: {
+                return richEmailBuilder.build(templateIdFromEvent(queryProcessingError),
+                                                                          Collections.emptyList(),
+                                                                          exception.toMap());
+            }
+            case TABLE_INITIALIZATION_ERROR:
+            case TABLE_METADATA_FETCH_FAILURE:
+            case TABLE_NOT_FOUND:
```

Comment:
maybe just replace all these error codes by default block since we are doing nothing for any other error code